### ISSUE TITLE
Updated user model not available in template when "change:logged_in" triggers render

### ISSUE THREAD

**Issue by jonmiles** [alexanderscott/backbone-login#3]
Firstly I'd like to thank you for your contribution.  This is a really great explanation of how to implement backbone login, and it was very useful for me.

However, I've noticed that the session.user is not updated correctly when the event 'change:logged_in' is handled by the view / template.

I have a slightly different setup than your example, I'm using Backbone Marionette and Backbone.Marionette.Handlebars helpers (hbs) to render / manage view templates.  What I noticed was that when the 'change:logged_in' event fires as a result of a successful user log in, that the user model when accessed via session.user was still the invalid user model.  It doesn't update correctly until the page is refreshed, which defeats the object!

My scenario is that I use the session model as the model of one of my views.  I have a handlebars template which looks first at 'logged_in', if this is true is looks to render the user's name using user.username.  My problem is that user.username is still the unauthenticated user response/ model.

I fixed my problem with the following change to postAuth function in SessionModel.js.

-self.updateSessionUser( res.user || {} );
-self.set({ user_id: res.user.id, logged_in: true });
+self.set({ 
      logged_in: true,
      user_id: res.user.id, 
      user: _.pick(res.user, _.keys(self.user.defaults))
    });

Basically if you set the user model as part of the session model set, the new user model is available via App.session.user, otherwise it is not.  Plus a similar change was needed in checkAuth for consistency.

I realise your example is structured slightly differently and what you have works for your example, but I think the user model should be set correctly at the point the session logged_in state is set.
